docs/serialization/v4.rst

```diff
@@ -21,10 +21,9 @@ We first define a set of enum types to be used in the serialization format.
 
   - ``kBinaryClf`` (0): binary classifier
   - ``kRegressor`` (1): regressor
-  - ``kLearningToRank`` (2): learning-to-rank
-  - ``kMultiClfGrovePerClass`` (3): multi-class classifier with grove-per-class
-  - ``kMultiClfProbDistLeaf`` (4): multi-class classifier with vector leaf outputs producing probability distribution
```

Comment:
Benefit of new `task_type`: By getting rid of `kMultiClfGrovePerClass`, we made the task type orthogonal to the rest of the model specification.

Previously, the task type imposed complex constraints on the model. For example:
https://github.com/dmlc/treelite/blob/3008035e34116e60e0e9a069c20ac6e346979a67/include/treelite/task_type.h#L47-L49

In the new approach, the task type no longer imposes such complex constraints.